dspace-spring-rest/src/main/java/org/dspace/app/rest/UUIDLookupRestController.java

```diff
@@ -0,0 +1,108 @@
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.TemplateVariable;
+import org.springframework.hateoas.TemplateVariable.VariableType;
+import org.springframework.hateoas.TemplateVariables;
+import org.springframework.hateoas.UriTemplate;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This is an utility endpoint to lookup a generic DSpaceObject. If found the controller will
+ * respond with a redirection to the canonical endpoint for the actual object.
+ *
+ * @author Andrea Bollini (andrea.bollini at 4science.it)
+ */
+@RestController
+@RequestMapping("/api/" + UUIDLookupRestController.CATEGORY)
+public class UUIDLookupRestController implements InitializingBean {
+    public static final String CATEGORY = "dso";
+
+    public static final String ACTION = "find";
+
+    public static final String PARAM = "uuid";
+
+    @Autowired
+    private ContentServiceFactory contentServiceFactory;
+
+    private static final Logger log =
+            Logger.getLogger(UUIDLookupRestController.class);
+
+    @Autowired
+    private DiscoverableEndpointsService discoverableEndpointsService;
+
+    @Autowired
+    private GenericDSpaceObjectConverter converter;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        discoverableEndpointsService
+            .register(this,
+                    Arrays.asList(
+                            new Link(
+                                    new UriTemplate("/api/" + CATEGORY + "/" + ACTION,
+                                            new TemplateVariables(
+                                                    new TemplateVariable(PARAM, VariableType.REQUEST_PARAM))),
+                                    CATEGORY)));
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = ACTION, params = PARAM)
+    @SuppressWarnings("unchecked")
+    public void getDSObyIdentifier(HttpServletRequest request,
+                                   HttpServletResponse response,
+                                   @RequestParam(PARAM) UUID uuid)
+            throws IOException, SQLException, SearchServiceException {
+
+        Context context = null;
+        try {
+            context = ContextUtil.obtainContext(request);
+            for (DSpaceObjectService dSpaceObjectService : contentServiceFactory.getDSpaceObjectServices()) {
+                DSpaceObject dso = dSpaceObjectService.find(context, uuid);
+                if (dso != null) {
+                    DSpaceObjectRest dsor = converter.convert(dso);
+                    URI link = linkTo(dsor.getController(), dsor.getCategory(), English.plural(dsor.getType()))
```

Comment:
Can you use `getTypePlural()` here? https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/model/RestModel.java#L35

Reply:
done